### PR TITLE
Tr 68

### PR DESCRIPTION
**Pull request by nazariykupchenko**

### CODE REVIEW

**Review by nazariykupchenko — COMMENTED** [review]
Should get rid of .idea folder with its files

File: src/main/java/com/api/endpoints/reqres/ReqresUserEndpoint.java [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+
+import static com.api.utils.Properties.REQRES_BASE_URI;
+import static com.api.utils.Properties.REQRES_USERS_END_POINT;
+
+import com.api.endpoints.Api;
+import io.restassured.response.ValidatableResponse;
+
+public class ReqresUserEndpoint extends Api {
+
+  public ValidatableResponse getSingleUser(int i) {
```
Comment (adzombak) [inline src/main/java/com/api/endpoints/reqres/ReqresUserEndpoint.java]: change variable name to id to be more clear about it

File: src/main/java/com/api/endpoints/swapi/SwapiUserEndpoint.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+
+import static com.api.utils.Properties.PEOPLE_END_POINT;
+import static com.api.utils.Properties.SWAPI_BASE_URI;
+
+import com.api.endpoints.Api;
+import io.restassured.response.ValidatableResponse;
+
+public class SwapiUserEndpoint extends Api {
+
+  public ValidatableResponse getUser(int i) {
```
Comment (adzombak) [inline src/main/java/com/api/endpoints/swapi/SwapiUserEndpoint.java]: swapi doesn't have users, update respectively

File: src/main/java/com/api/models/reqres/ListOfResourcesModel.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  private String name;
+  @JsonProperty("color")
+  private String color;
+  @JsonProperty("pantone_value")
+  private String pantone_value;
+  @JsonProperty("url")
+  private String url;
+  @JsonProperty("text")
+  private String text;
+
```
Comment (adzombak) [inline src/main/java/com/api/models/reqres/ListOfResourcesModel.java]: remove unnecessary lines

File: src/main/java/com/api/models/reqres/SingleResourceModel.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  private int year;
+  @JsonProperty("color")
+  private String color;
+  @JsonProperty("pantone_value")
+  private String pantone_value;
+  @JsonProperty("url")
+  private String url;
+  @JsonProperty("text")
+  private String text;
+
```
Comment (adzombak) [inline src/main/java/com/api/models/reqres/SingleResourceModel.java]: in all such files

**Review by adzombak — CHANGES_REQUESTED** [review]
update code formatting and remove idea files from git

File: src/main/java/com/api/models/reqres/ListOfResourcesModel.java (hunk 3 above)
Comment (adzombak) [reply src/main/java/com/api/models/reqres/ListOfResourcesModel.java]: @nazariykupchenko in all models after the last variable and '}' all lines should be removed

File: src/main/java/com/api/endpoints/Api.java [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+    RestAssured.baseURI = baseUri;
+    return RestAssured
+        .given()
+        .header("Content-Type", ContentType.JSON)
+        .header("connection", "keep-alive")
+        .log()
+        .ifValidationFails();
+//        .filters(new RequestLoggingFilter(), new ResponseLoggingFilter());
+  }
+
```
Comment (adzombak) [inline src/main/java/com/api/endpoints/Api.java]: this line should be removed

File: src/main/java/com/api/models/reqres/CreateUserModel.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+  @JsonProperty("id")
+  private int id;
+  @JsonProperty("name")
+  private String name;
+  @JsonProperty("job")
+  private String job;
+  @JsonProperty("createAt")
+  private String createAt;
+
```
Comment (adzombak) [inline src/main/java/com/api/models/reqres/CreateUserModel.java]: this line should be removed

File: src/main/java/com/api/models/reqres/ListOfUsersModel.java [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+  private int id;
+  @JsonProperty("email")
+  private String email;
+  @JsonProperty("first_name")
+  private String first_name;
+  @JsonProperty("last_name")
+  private String last_name;
+  @JsonProperty("avatar")
+  private String avatar;
+
```
Comment (adzombak) [inline src/main/java/com/api/models/reqres/ListOfUsersModel.java]: this line should be removed

File: src/main/java/com/api/models/reqres/SingleResourceModel.java [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+  private int id;
+  @JsonProperty("name")
+  private String name;
+  @JsonProperty("year")
+  private int year;
+  @JsonProperty("color")
+  private String color;
+  @JsonProperty("pantone_value")
+  private String pantone_value;
+
```
Comment (adzombak) [inline src/main/java/com/api/models/reqres/SingleResourceModel.java]: this line should be removed

File: src/main/java/com/api/models/reqres/SingleUserModel.java [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+  private int id;
+  @JsonProperty("email")
+  private String email;
+  @JsonProperty("first_name")
+  private String first_name;
+  @JsonProperty("last_name")
+  private String last_name;
+  @JsonProperty("avatar")
+  private String avatar; //тут ссилка на img
+
```
Comment (adzombak) [inline src/main/java/com/api/models/reqres/SingleUserModel.java]: this line should be removed

File: src/main/java/com/api/models/reqres/UpdateUserModel.java [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+@Getter
+public class UpdateUserModel {
+
+  @JsonProperty("name")
+  private String name;
+  @JsonProperty("job")
+  private String job;
+  @JsonProperty("updateAt")
+  private String updateAt;
+
```
Comment (adzombak) [inline src/main/java/com/api/models/reqres/UpdateUserModel.java]: this line should be removed

File: src/main/java/com/api/models/reqres/CreateUserModel.java (hunk 6 above)
Comment (nazariykupchenko) [reply src/main/java/com/api/models/reqres/CreateUserModel.java]: @adzombak , а чому createAt видаляти?

File: src/main/java/com/api/models/reqres/SingleUserModel.java [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+  @JsonProperty("id")
+  private int id;
+  @JsonProperty("email")
+  private String email;
+  @JsonProperty("first_name")
+  private String first_name;
+  @JsonProperty("last_name")
+  private String last_name;
+  @JsonProperty("avatar")
+  private String avatar; //тут ссилка на img
```
Comment (adzombak) [inline src/main/java/com/api/models/reqres/SingleUserModel.java]: don't leave such comments

File: src/main/java/com/api/endpoints/Api.java (hunk 5 above)
Comment (adzombak) [inline src/main/java/com/api/endpoints/Api.java]: this line is not removed

**Review by adzombak — APPROVED** [review]

**Pull request merged**